### DISCUSSION TITLE
Prebuilt clang does not have llvm core library(dll) for windows.

### DISCUSSION THREAD

**Discussion by codetronik** [android/ndk#1922] [Q&A] [answered]
Questions may not be appropriate for this, but they are related to ndk deployment.

I'm trying to build llvm pass for ndk with prebuilt clang.
For linux, we provide llvm library, but Windows doesn't have dll, so it's impossible to build a pass.
Even darwin includes dylib library.
It was built easily on Linux and mac.
Not for Windows in the plan?

you can check it at the link below.

Darwin prebuilt clang
https://android.googlesource.com/platform/prebuilts/clang/host/darwin-x86/+/refs/heads/ndk-r25-release/clang-r450784d1/lib64/

Windows prebuilt clang
https://android.googlesource.com/platform/prebuilts/clang/host/windows-x86/+/refs/heads/ndk-r25-release/clang-r450784d1/lib64/

Linux prebuilt clang
https://android.googlesource.com/platform/prebuilts/clang/host/linux-x86/+/refs/heads/ndk-r25-release/clang-r450784d1/lib64/

**Comment by DanAlbert** [comment, marked as answer]
Not something the NDK supports. You need to build your own toolchain. https://android.googlesource.com/toolchain/llvm_android/+/refs/heads/main/README.md has instructions for rebuilding our toolchain.

**Comment by codetronik** [comment]
It doesn't work either. There is no code in do_build.py to create .dll.
```
def normalize_llvm_host_libs(install_dir: Path, host: hosts.Host, version: Version) -> None:
    if host.is_linux:
        libs = {'libLLVM': 'libLLVM-{version}git.so',
                'libclang': 'libclang.so.{version}git',
                'libclang-cpp': 'libclang-cpp.so.{version}git',
                'libc++': 'libc++.so.{version}',
                'libc++abi': 'libc++abi.so.{version}'
               }
    else:
        libs = {'libc++': 'libc++.{version}.dylib',
                'libc++abi': 'libc++abi.{version}.dylib'
               }
```

**Reply by DanAlbert** [reply]
Send them a patch or ask on https://groups.google.com/g/android-llvm. I can't help you.
